tests/framework/test_time.py

```diff
@@ -56,94 +62,106 @@ def step_column(sim):
+
+
+def test_basic_iteration(SimulationContext, base_config, components):
+    base_config["time"]["step_size"] = 1
     sim = SimulationContext(base_config, components)
+    listener = [c for c in components if "listener" in c.args][0]
     sim.setup()
     sim.initialize_simulants()
+    assert sim._clock.time == get_time_stamp(sim.configuration.time.start)
+    assert sim._clock.step_size == pd.Timedelta(days=1)
     pop_size = len(sim.get_population())
     # After initialization, all simulants should be aligned to event times
     assert len(active_simulants(sim)) == pop_size
+    assert np.all(step_pipeline(sim) == step_column(sim))
 
-    sim.step()
-    # After one step (and no step adjustments, simulants should still be aligned)
-    assert len(active_simulants(sim)) == pop_size
-
-    sim._population._population.step_size *= 2
-
-    sim.step()
-    # No simulants should be aligned after a step size adjustment
-    assert active_simulants(sim).empty
-
-    sim.step()
-    # Now they should be aligned again
-    assert len(active_simulants(sim)) == pop_size
+    for _ in range(2):
+        taken_step_size = take_step(sim)
+        # After a step (and no step adjustments, simulants should still be aligned)
+        assert taken_step_size == pd.Timedelta(days=1)
+        assert len(active_simulants(sim)) == pop_size
+        assert np.all(step_pipeline(sim) == step_column(sim))
+        for index in listener.event_indexes.values():
+            assert np.all(index == sim.get_population().index)
 
 
-def test_unequal_steps(SimulationContext, base_config, components):
+def test_empty_active_pop(SimulationContext, base_config, components):
+    base_config["time"]["step_size"] = 1
     sim = SimulationContext(base_config, components)
     listener = [c for c in components if "listener" in c.args][0]
     sim.setup()
     sim.initialize_simulants()
     pop_size = len(sim.get_population())
 
-    # Check that the 0th simulant won't step forward
-    sim._population._population.step_size[0] *= 2
-    sim.step()
-    assert len(active_simulants(sim)) == pop_size - 1
+    ## Force a next event time update without updating step sizes.
+    ## This ensures (against the current implementation) that we will have a timestep
+    ## that has no simulants aligned. Check that we do the minimum timestep update.
+    sim._population._population.next_event_time += pd.Timedelta(days=1)
 
-    # Show that now that the next_event_time is updated, 0th simulant isn't included in events
-    sim.step()
+    assert active_simulants(sim).empty
+    taken_step_size = take_step(sim)
+    assert taken_step_size == pd.Timedelta(days=1)
     for index in listener.event_indexes.values():
-        assert 0 not in index
+        assert index.empty
 
-    # Check that everybody will step forward next step
     assert len(active_simulants(sim)) == pop_size
-
-    # Check that they are actually included in events
-    sim.step()
+    taken_step_size = take_step(sim)
```

Comment:
Shouldn't active_simulants be empty here on the line above? You previously took a time step and `assert index.empty`.

Reply:
The `active_simulants` method reutrns the indices of the simulants that are supposed to be included *before* timestep/lifecycle events are emitted. The `Listener` mock component stores the index acted upon *most recently* by timestep/liecycle events that were previously emitted. So here we are checking that before an event, we're "loading" the right simulants, as it were, and then, after doing the timestep, we verify that the same simulants were in fact updated. After the timestep has been taken, the `active_simulants` is now the group of simulants that will be updated upon the next iteration of events.